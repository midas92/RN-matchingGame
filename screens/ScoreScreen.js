import React, { useState, useEffect } from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
import { storeData, getData } from '../storage/DataStorage';
import { DataTable } from 'react-native-paper';
import { dummyScoreData } from '../storage/SeedData';

const ScoreScreen = () => {
  const [scores, setScores] = useState([]);
  const [page, setPage] = useState(0);
  const RESULTS_PER_PAGE = 10;

  useEffect(() => {
    getData('@scores').then(result => setScores(result));
  }, []);

  useEffect(() => {
    setPage(Math.ceil(scores.length / RESULTS_PER_PAGE) - 1);
  }, [scores]);

  const setupScores = () => {
    storeData('@scores', dummyScoreData());
  }

  return (
    <View style={styles.gameContainer}>
      
      <View style={styles.textContainer}>
        <Text style={styles.textEntity}>High Scores</Text>
        {/* <TouchableOpacity onPress={() => setupScores()}>
          <Text> Setup Scores</Text>
        </TouchableOpacity> */}
      </View>

      <View style={styles.tableContainer}>
        
        <DataTable>
          <DataTable.Header>
            <DataTable.Title style={styles.nameCell}>Name</DataTable.Title>
            <DataTable.Title style={styles.centerCell}>Matrix</DataTable.Title>
            <DataTable.Title style={styles.centerCell}>Matches</DataTable.Title>
            <DataTable.Title numeric style={styles.centerCell}>Score</DataTable.Title>
            <DataTable.Title style={styles.dateCell}>Date</DataTable.Title>
          </DataTable.Header>

          { scores.slice(page * RESULTS_PER_PAGE, (page + 1) * RESULTS_PER_PAGE).map((value, index) => {
            const { name, matrix, numberOfMatches, score, date } = value;
            const scoreDate = new Date(date);
            
            return (
              <DataTable.Row key={index}>
                <DataTable.Cell style={styles.nameCell}>
                  <Text>{name}</Text>
                </DataTable.Cell>
                <DataTable.Cell style={styles.centerCell}>
                  <Text>{matrix}</Text>
                </DataTable.Cell>
                <DataTable.Cell style={styles.centerCell}>
                  <Text>{numberOfMatches}</Text>
                </DataTable.Cell>
                <DataTable.Cell numeric style={styles.centerCell}>
                  <Text>{score}</Text>
                </DataTable.Cell>
                <DataTable.Cell style={styles.dateCell}>
                  {scoreDate.toLocaleDateString()} ({scoreDate.getHours()}:{scoreDate.getMinutes()})
                </DataTable.Cell>
              </DataTable.Row>
            )          
          })}

          <DataTable.Pagination
            page={page}
            numberOfPages={Math.ceil(scores.length / RESULTS_PER_PAGE)}
            onPageChange={page => setPage(page)}
            label={`Page ${(page + 1)} of ${Math.ceil(scores.length / RESULTS_PER_PAGE)}`}
            />
        </DataTable>
      </View>
    </View>
  );
}

export default ScoreScreen;
  
const styles = StyleSheet.create({
  gameContainer: {
    flex: 1,
    backgroundColor: '#E3E3E3',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textContainer: {
    flex: 1,
    paddingTop: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  textBox: {
    width: '33.33%',
    alignItems: 'center',
  },
  textEntity: {
    fontSize: 20,
  },
  tableContainer: {
    flex: 8,
    alignItems: 'center',
    width: '95%'
  },
  resetButton: {
    height: 40,
    width: 400,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: 'black',
    shadowRadius: 2,
    shadowOpacity: 1,
    backgroundColor: '#34a8eb',
    borderRadius: 50,
    flexDirection: 'row',
    marginTop: 10
  },
  resetIcon: {
    transform: [
      { scaleX: -1 }
    ],
  },
  resetButtonText: {
    fontSize: 20,
    marginRight: 10
  },
  nameCell: {
    flex: 7,
  },
  centerCell: {
    justifyContent: 'center',
    flex: 5,
  },
  dateCell: {
    justifyContent: 'flex-end',
    flex: 10
  }
})
