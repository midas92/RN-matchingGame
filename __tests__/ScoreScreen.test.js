import React from 'react';
import { act, render } from '@testing-library/react-native';
import ScoreScreen from '../screens/ScoreScreen';

// Need to figure out how to work with the child component - DataTable. Currently is mocked out

describe('<ScoreScreen/>', () => {
  let result;
  beforeEach(() => result = render(<ScoreScreen/>));

  it('should match snapshot', async () => {
    await act(async () => {
      expect(result.toJSON()).toMatchSnapshot()
    })
  });
  it('should render the ScoreScreen component', () => expect(result).toBeTruthy());
});
