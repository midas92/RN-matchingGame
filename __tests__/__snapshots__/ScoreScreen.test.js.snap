// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoreScreen/> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E3E3E3",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      High Scores
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 8,
        "width": "98%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 48,
              "paddingHorizontal": 16,
            },
            Object {
              "borderBottomColor": "rgba(0, 0, 0, 0.12)",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignContent": "center",
                "flex": 1,
                "flexDirection": "row",
                "paddingVertical": 12,
              },
              undefined,
              Object {
                "flex": 6,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "height": 24,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.6)",
                  },
                ],
              ]
            }
          >
            Name
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignContent": "center",
                "flex": 1,
                "flexDirection": "row",
                "paddingVertical": 12,
              },
              undefined,
              Object {
                "flex": 5,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "height": 24,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.6)",
                  },
                ],
              ]
            }
          >
            Matrix
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignContent": "center",
                "flex": 1,
                "flexDirection": "row",
                "paddingVertical": 12,
              },
              undefined,
              Object {
                "flex": 5,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "height": 24,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.6)",
                  },
                ],
              ]
            }
          >
            Matches
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignContent": "center",
                "flex": 1,
                "flexDirection": "row",
                "paddingVertical": 12,
              },
              Object {
                "justifyContent": "flex-end",
              },
              Object {
                "flex": 5,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "height": 24,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.6)",
                  },
                ],
              ]
            }
          >
            Score
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignContent": "center",
                "flex": 1,
                "flexDirection": "row",
                "paddingVertical": 12,
              },
              undefined,
              Object {
                "flex": 10,
                "justifyContent": "flex-end",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "height": 24,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.6)",
                  },
                ],
              ]
            }
          >
            Date
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "paddingLeft": 16,
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 12,
                  "marginRight": 44,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.6)",
                },
              ],
            ]
          }
        >
          Page 1 of 0
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "height": 36,
                  "width": 36,
                },
                Object {
                  "opacity": 0.32,
                },
                undefined,
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                ]
              }
            >
              □
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "height": 36,
                  "width": 36,
                },
                Object {
                  "opacity": 0.32,
                },
                undefined,
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                ]
              }
            >
              □
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
